packages/lexical-editor-pb/src/plugins/PeTextPlugin.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useMemo } from "react";
+import PeText from "@webiny/app-page-builder/editor/components/Text/PeText";
+import { PeText as LexicalPeText } from "~/components/PeText";
+import { createComponentPlugin } from "@webiny/react-composition";
+import { useRecoilState, useRecoilValue } from "recoil";
+import {
+    elementWithChildrenByIdSelector,
+    uiAtom
+} from "@webiny/app-page-builder/editor/recoil/modules";
+import get from "lodash/get";
+import { applyFallbackDisplayMode } from "@webiny/app-page-builder/editor/plugins/elementSettings/elementSettingsUtils";
+import { isParagraphTag } from "~/utils/isParagraphTag";
+import { isHeadingTag } from "~/utils/isHeadingTag";
+
+const DATA_NAMESPACE = "data.text";
+export const PeTextPlugin = createComponentPlugin(PeText, Original => {
+    return function PbTextPlugin({ elementId, tag: customTag, mediumEditorOptions }): JSX.Element {
+        const element = useRecoilValue(elementWithChildrenByIdSelector(elementId));
+        const [{ displayMode }] = useRecoilState(uiAtom);
```

Comment:
There are hooks for this, so you don't have to work with recoil directly, check this folder and simplify your code usign these hooks: https://github.com/webiny/webiny-js/tree/dev/packages/app-page-builder/src/editor/hooks